src/ai/backend/storage/cephfs/__init__.py

```diff
@@ -0,0 +1,87 @@
+import asyncio
+from typing import Dict, List
+from uuid import UUID
+
+from ai.backend.common.types import BinarySize
+
+from ..exception import ExecutionError
+from ..types import FSUsage, VFolderCreationOptions
+from ..vfs import BaseVolume
+
+
+async def read_file(loop: asyncio.AbstractEventLoop, filename: str) -> str:
+    def _read():
+        with open(filename, "r") as fr:
+            return fr.read()
+
+    return await loop.run_in_executor(None, lambda: _read())
+
+
+async def write_file(loop: asyncio.AbstractEventLoop, filename: str, contents: str, perm="w"):
+    def _write():
+        with open(filename, perm) as fw:
+            fw.write(contents)
+
+    await loop.run_in_executor(None, lambda: _write())
+
+
+async def run(cmd: str) -> str:
+    proc = await asyncio.create_subprocess_shell(
+        cmd, stdout=asyncio.subprocess.PIPE, stderr=asyncio.subprocess.PIPE
+    )
+    out, err = await proc.communicate()
+    if err:
+        raise ExecutionError(err.decode())
+    return out.decode()
+
+
+class CephFSVolume(BaseVolume):
+    loop: asyncio.AbstractEventLoop
+    registry: Dict[str, int]
+    project_id_pool: List[int]
+
+    async def init(self) -> None:
+        available = True
+        try:
+            await asyncio.create_subprocess_exec(
+                b"ceph",
+                b"--version",
+                stdout=asyncio.subprocess.PIPE,
+                stderr=asyncio.subprocess.STDOUT,
+            )
+        except FileNotFoundError:
+            available = False
+
+        if not available:
+            raise RuntimeError("Ceph is not installed. ")
+
+    # ----- volume opeartions -----
+    async def create_vfolder(self, vfid: UUID, options: VFolderCreationOptions = None) -> None:
+        vfpath = self.mangle_vfpath(vfid)
+        await run(f"mkdir -p {vfpath}")
```

Comment:
`BaseVolume.create_vfolder()`, overridden method of this method, does exactly same thing with these lines. I think it's be better to let them do its job.
```suggestion
        await super().create_vfolder(vfid, options, exist_ok=exist_ok)
```